lms/djangoapps/courseware/access.py

```diff
@@ -253,6 +253,21 @@ def _can_enroll_courselike(user, courselike):
     # which actually points to a CourseKey. Sigh.
     course_key = courselike.id
 
+    if settings.FEATURES.get('RESTRICT_ENROLL_SOCIAL_PROVIDERS'):
+        try:
+            social_providers = user.social_auth.all()
+        except AttributeError:
+            social_providers = []
+        allowed_providers = settings.FEATURES.get('RESTRICT_ENROLL_SOCIAL_PROVIDERS')
+        if social_providers and not any(
+            [any([social_provider.uid.startswith(allowed_provider)
+                  for allowed_provider in allowed_providers])
+             for social_provider in social_providers]):
+            log.warning(
+                "ENROLLMENT DENIED in %s because user %s not authenticated by any of %s",
+                course_key, user.username, allowed_providers)
+            return ACCESS_DENIED
+
```

Comment:
This change seems out of place. I think this functionality should already be part of lilac, but possibly under a different setting name. See https://github.com/edx/edx-platform/pull/24789

Reply:
As per my understadning this seems like allowing particular social providers from the list of available.

But edx#24789 seems something different to only allowing third party authentication and blocking the normal flow of authentication.

@pdpinch What is your final opinion on that?

Reply:
@mitodl/devops do we need this setting for the residential deployment? 

It looks like it's set in [salt-ops](https://github.com/mitodl/salt-ops/blob/3aa25b111492f7b8f50e552189127f81a6d9f497/pillar/edx/ansible_vars/residential.sls#L175-L176)

@umarmughal824 if we need it, we should keep it in the cherry-pick -- and we should consider upstreaming it to edx.

Reply:
If we do need to keep this, perhaps it should be in a separate PR, like we did for koa in https://github.com/mitodl/edx-platform/pull/228

Reply:
So, this setting is for allowing us to be able to gate enrollments from touchstone collaborators. There is still the open question about whether collaborator accounts are even a thing any more, so we'll likely need to do some investigation about how to manage that restriction going forward.